cmd/lib.go

```diff
@@ -36,18 +33,31 @@ var defaultStoreHostTargetPath = "/var/lib/resonance"
 
 func wrapHost(ctx context.Context, hst host.Host) (host.Host, error) {
 	var err error
-	if sudo {
-		hst, err = ihost.NewSudo(ctx, hst)
-		if err != nil {
-			return nil, err
-		}
+
+	optionsMap := map[string]bool{
+		"sudo":          false,
+		"disable-agent": false,
 	}
+	if options != "" {
+		for _, o := range strings.Split(options, ",") {
+			if _, ok := optionsMap[o]; !ok {
+				return nil, fmt.Errorf("invalid option: %#v", o)
+			}
+			optionsMap[o] = true
+		}
 
-	if !disableAgent && ssh != "" {
-		var err error
-		hst, err = ihost.NewAgent(ctx, hst)
-		if err != nil {
-			return nil, err
+		if optionsMap["sudo"] {
+			hst, err = ihost.NewSudo(ctx, hst)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		if hst.Type() != "localhost" && !optionsMap["disable-agent"] {
+			hst, err = ihost.NewAgent(ctx, hst)
+			if err != nil {
+				return nil, err
+			}
 		}
 	}
```

Comment:
@gfechio I'm really sorry to iterate this one more time, but I think you'll likely agree this is important. I hope I haven't missed anything here, if I did, **please** let me know (I'm sorry!). I recokn this is frustrating, and I promisse to only focus on important comments on future PRs: I'm sorry for what happened here.

If you check this comment: https://github.com/fornellas/resonance/pull/77/files#r1725719671, I suggested puling this outside of the `if options != ""` clause, because otherwise, when no option is passed, it'd _not_ be enabling the agent. We most definitely don't want that, as the agent can make things 25-90x faster (see benchmarks on #2). We touched the subject on a previous comment https://github.com/fornellas/resonance/pull/77#discussion_r1717481605 as well, where I explained the cases where the agent must be enabled.

WDYT?

```suggestion
	}
	if hst.Type() != "localhost" && !optionsMap["disable-agent"] {
		hst, err = ihost.NewAgent(ctx, hst)
		if err != nil {
			return nil, err
		}
	}
```

Reply:
That makes sense, I think I might have missed it. :smile: